### PR TITLE
closes #17. Tag enforcement

### LINKED ISSUE #17: Tag enforcement (sites without tags don't appear)
Current logic does not show sites if there isn't a tag associated with the site.

Comments from @hackertron

"1. Yes tags should be mandatory, i don't remember if i enforced that on schema
2. Frontend and backend both should have enforced that. Frontend should not have allowed form submission and backend should not have accepted it
3.. We can add tags yes by directly modifying the db. (should not but will)
4. Users won't be able to do that. Once url is submitted it's submitted"

"there are total 12 url submission where there are no tags, so they won't be fetched ever if user selects tags and discover.  They will only be shown when no tags are selected."

From @thebeyondr 

"Ok I changed the query to find() in the cntrllr, I'll try to update so it fetches for things with atleast one tag. We should tag the untagged ones. Also I'll convert to lowercase before saving tags."

### PR DESCRIPTION
**Pull request by hackertron**

### CODE REVIEW

**Comment by tpantpan** [comment]
No 'submit tag' page anymore, tag creation capability will be part of the 'submit site' page. Unable to create tags just by themselves.

Something to consider: Will not be able to submit website without a tag, but want to make it more friendly (design preference is not to disable features). Will need to do some more design research on this.

@thebeyondr please create a new issue for this

File: controllers/urls.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ const createURL = async (req, res) => {
   try {
     const [title, url, tags] = req.body.params;
     const submittedBy = req.body.userId;
+
+    if (!tags || tags.length === 0) {
+      return res.status(400).json({ error: "Tags cannot be empty" });
```
Comment (thebeyondr) [inline controllers/urls.js]:
```suggestion
      return res.status(400).json({ error: "Please add some tags to the url" });
```

**Review by thebeyondr — APPROVED** [review]
Ez but I suggested a more friendly error message

**Review by NicoSerranoP — APPROVED** [review]

**Comment by thebeyondr** [comment]
> @thebeyondr please create a new issue for this

Done: https://github.com/State-Channel-4/frontend/issues/25

**Pull request merged**